scripts/isolate_tests.py

```diff
@@ -35,98 +37,127 @@ def extract_test_cases(path):
+    ]
+
+# Extract code examples based on the 'beginMarker' parameter
+# up until we reach EOF or a line that is not empty and doesn't start with 4
+# spaces.
+def extract_docs_cases(path, beginMarker):
+    immediatelyAfterMarker = False
     insideBlock = False
-    insideBlockParameters = False
-    pastBlockParameters = False
-    extractedLines = []
     tests = []
 
     # Collect all snippets of indented blocks
-
     with open(path, mode='r', errors='ignore', encoding='utf8', newline='') as f:
         lines = f.read().splitlines()
-    for l in lines:
-        if l != '':
-            if not insideBlock and l.startswith(' '):
-                # start new test
-                extractedLines += ['']
-                insideBlockParameters = False
-                pastBlockParameters = False
-            insideBlock = l.startswith(' ')
-        if insideBlock:
-            if not pastBlockParameters:
-                # NOTE: For simplicity this allows blank lines between block parameters even
-                # though Sphinx does not. This does not matter since the first non-empty line in
-                # a Solidity file cannot start with a colon anyway.
-                if not l.strip().startswith(':') and (l != '' or not insideBlockParameters):
-                    insideBlockParameters = False
-                    pastBlockParameters = True
-                else:
-                    insideBlockParameters = True
-
-            if not insideBlockParameters:
-                extractedLines[-1] += l + '\n'
-
-    codeStart = "(// SPDX-License-Identifier:|pragma solidity|contract.*{|library.*{|interface.*{)"
 
-    # Filter all tests that do not contain Solidity or are indented incorrectly.
-    for lines in extractedLines:
-        if re.search(r'^\s{0,3}' + codeStart, lines, re.MULTILINE):
-            print("Indentation error in " + path + ":")
-            print(lines)
-            exit(1)
-        if re.search(r'^\s{4}' + codeStart, lines, re.MULTILINE):
-            tests.append(lines)
+    for line in lines:
+        if insideBlock:
+            if immediatelyAfterMarker:
+                # Skip Shpinx instructions and empty lines between them
+                if line == '' or line.lstrip().startswith(":"):
+                    continue
+
+            if line == '' or line.startswith("    "):
+                tests[-1] += line + "\n"
+                immediatelyAfterMarker = False
+            else:
+                insideBlock = False
+        elif line.lower().startswith(beginMarker):
+            insideBlock = True
+            immediatelyAfterMarker = True
+            tests += ['']
 
     return tests
 
-def write_cases(f, tests):
+def write_cases(f, tests, yulTests):
     cleaned_filename = f.replace(".","_").replace("-","_").replace(" ","_").lower()
-    for test in tests:
+    for lang, test in chain(
+        map(lambda t: ("sol", t), tests),
+        map(lambda t: ("yul", t), yulTests)
+    ):
```

Comment:
Your version is fine but it can be made simpler:
```suggestion    
    for lang, test in [("sol", t) for t in tests] + [("yul", t) for t in yulTests]:
```